homeassistant/setup.py

```diff
@@ -217,9 +217,24 @@ def log_error(msg: str, exc_info: Exception | None = None) -> None:
         log_error(f"Unable to import component: {err}", err)
         return False
 
-    processed_config = await conf_util.async_process_component_config(
-        hass, config, integration
-    )
+    (
+        processed_config,
+        config_exceptions,
+    ) = await conf_util.async_process_component_config(hass, config, integration)
+    for ex_info in config_exceptions:
+        if ex_info.log_message is not None:
+            _LOGGER.error(
+                ex_info.log_message,
+                exc_info=ex_info.ex if ex_info.log_exception else None,
+            )
+            continue
+        conf_util.async_log_config_validator_error(
+            ex_info.ex,
+            ex_info.platform_name,
+            ex_info.config,
+            hass,
+            ex_info.integration_link or integration.documentation,
+        )
```

Comment:
What's happening here? Why do we use a different method for iterating over the errors which doesn't get the file/line info and so on?